### ISSUE TITLE
.sbtopts temporary removal suppresses sbt.repositories

### ISSUE THREAD

**Issue by Slakah** [fthomas/scala-steward#365]
A common pattern in sbt is to override the repositories, by specifying the following in the `.sbtopts` of a project.

i.e.
`.sbtopts`:
```
-Dsbt.override.build.repos=true
-Dsbt.repository.config=sbt.repositories
```
The following code temporarily removes the `.sbtopts` file.
https://github.com/fthomas/scala-steward/blob/bf3191ed44ce9e39de7195b64b0bc2ff09c8e6ac/modules/core/src/main/scala/org/scalasteward/core/sbt/SbtAlg.scala#L107

I can't think of a sensible solution.

Maybe we could add a whitelist for certain params?

I suppose a workaround, would be to define resolvers inside the build sbt definitions.

Thanks for the great tool!

**Comment by fthomas** [comment]
I'm removing `.sbtopts` and `.jvmopts` because some projects use them to set `-Xms` values which are too high for my server. So these files are only removed because my server has not enough memory.

One easy way to keep these files when running scala-steward yourself would be to introduce a new config option which would change the behaviour of `ignoreOptsFiles`.

**Comment by Slakah** [comment]
I've had a quick think about it, would you be open to a PR which would do the following?
* Introduce `FileAlg#editFileTemporarily`.
* Produce a whitelist of SBT_OPT flags, probably `-Dsbt.override.build.repos` and `Dsbt.repository.config`.
* Change `ignoreOptsFiles`, honour the SBT_OPT whitelist.

Another option might be to instead of introducing `editFileTemporarily`:
* Read the `.sbtopts` before it's temporarily removed.
* Extract whitelisted SBT_OPT flags.
* Supply whitelisted SBT_OPT flags to `SbtAlg#sbtCmd`.

I can't promise if/when I might find time to raise such a PR.

Again thanks for the help!

**Comment by fthomas** [comment]
I would prefer to not parse `.sbtopts` files if not absolutely necessary. A new config option that disables `ignoreOptsFiles` would be much simpler. But maybe I don't understand your motivation for this change. Do you need this for a company internal instance of scala-steward you are running yourself or do you want that my public instance honors some flags defined in `.sbtopts`?

**Comment by Slakah** [comment]
> I would prefer to not parse .sbtopts files if not absolutely necessary. A new config option that disables ignoreOptsFiles would be much simpler.

That's fair, I'll try and find time to raise a PR.

> Do you need this for a company internal instance of scala-steward you are running yourself or do you want that my public instance honors some flags defined in .sbtopts?

Yep.

**Comment by fthomas** [comment]
Fixed by #380.
